### PR TITLE
Fix linker errors and RELEASE.local for the test IOC

### LINKED ISSUE #30: multiple definition errors when building with C++20
I'm using C++20 and when linking the source files, there a numerous multiple definition errors due to defining variables in the header files.

### PR DESCRIPTION
**Pull request by JJL772**
Fixes some issues I ran into when building this locally.

Closes #30

### CODE REVIEW

**Review by marciodo — COMMENTED** [review]
Using extern in variables inside header files doesn't seem right to me. Can you comment more on it? Maybe add the compiler message error that you are getting without it?

File: iocs/ipmicomm-test-IOC/configure/RELEASE
Comment (marciodo) [inline iocs/ipmicomm-test-IOC/configure/RELEASE:1]: Why changing the location of the includes changed the compiler behavior? We didn't see any issue with this before. Can you paste the error messages you've seen?

**Comment by JJL772** [comment]
The IOC specifically fails to build with linker errors. This is due to how static libraries are generated; each translation unit that includes a header containing a variable definition without external or static linkage (i.e. `int test;`) will get a globally visible definition. When the linker assembles them, it sees more than one symbol with the name and gets upset.

image

**Comment by JJL772** [comment]
This is actually a symptom of `-fcommon` and `-fno-common`. It links fine if you force the compiler to emit everything into the same data/bss section with `-fcommon`. GCC docs say `-fno-common` is the default though: https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#index-fcommon

> The -fcommon places uninitialized global variables in a common block. This allows the linker to resolve all tentative definitions of the same variable in different compilation units to the same object, or to a non-tentative definition. This behavior is inconsistent with C++, and on many targets implies a speed and code size penalty on global variable references. It is mainly useful to enable legacy code to link without errors. 

The default probably changed since GCC 4.8.5.

**Comment by gustavosr8** [comment]
> The IOC specifically fails to build with linker errors. This is due to how static libraries are generated; each translation unit that includes a header containing a variable definition without external or static linkage (i.e. `int test;`) will get a globally visible definition. When the linker assembles them, it sees more than one symbol with the name and gets upset.
> 
> image

I've been through the same problems and it worked fine with the changes in @JJL772's branch.

File: iocs/ipmicomm-test-IOC/configure/RELEASE
Comment (marciodo) [reply iocs/ipmicomm-test-IOC/configure/RELEASE:1]: @JJL772, I'd like to understand the change in configure/RELEASE before approving this pull request.

**Review by marciodo — APPROVED** [review]
Run in the test environment and it worked.

**Pull request merged**